platforms/core-runtime/process-services/src/main/java/org/gradle/process/internal/ClientExecHandleFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.internal.service.scopes.Scope;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+@NonNullApi
+@ServiceScope({Scope.Global.class, Scope.BuildSession.class})
+public interface ClientExecHandleFactory {
+    ClientExecHandleBuilder newExec();
```

Comment:
:x: I think we should not abbreviate here. How about `createExecHandleBuilder()` or something?

Reply:
Makes sense, yes. I agree it's not obvious when you see just `newExec` in the code what that returns, done